### DISCUSSION TITLE
Incrementing attributes of Nodes

### DISCUSSION THREAD

**Discussion by devjk1** [ueberdosis/tiptap#2326] [General Questions] [answered]
Any tips on how to create Paragraph nodes with incrementing ids/attributes?  The ids would increment for each line roughly, like `editor.getJSON().content.length`
I was thinking perhaps I could configure the Paragraph extension to include a dynamic "id", where the id would equal `editor.getJSON().content.length` but the editor isn't available yet in mounted
```
mounted() {
    this.editor = new Editor({
        content: this.modelValue,
        extensions: [
            Paragraph.configure({
                HTMLAttributes: {
                    id: this.editor.getJSON().content.length
                }
            }),
```
Last resort would be to recreate the HTML manually on submit, but perhaps there is a better way.

Thanks for your time and appreciate any ideas.

**Comment by hanspagel** [comment, marked as answer]
You can use the Table of Content example as a starting point.

https://tiptap.dev/guide/node-views/examples#table-of-contents
